addons/storyquest_bootstrap/plugin.gd

```diff
@@ -0,0 +1,167 @@
+
+const NEW_STORYQUEST_DIALOG = preload(
+	"res://addons/storyquest_bootstrap/new_storyquest_dialog.tscn"
+)
+
+const TOOL_MENU_LABEL := "Create StoryQuest from template..."
+
+const STORYQUESTS_PATH := "res://scenes/quests/story_quests/"
+const TEMPLATE_PATH := STORYQUESTS_PATH + "template/"
+const MIN_TITLE_LENGTH := 4
+
+const NEXT_SCENE_FORMAT = 'next_scene = "%s")'
+const NEXT_SCENE_PATTERN = 'next_scene = "(?<next_scene_uid>[^\\s]*")'
+
+static var _next_scene_regex := RegEx.create_from_string(NEXT_SCENE_PATTERN)
+
+var template_info := {
+	"intro":
+	{
+		"scene_path": "/0_template_intro/template_intro.tscn",
+		"components_path": "/0_template_intro/template_intro_components/",
+		"dialogue_path": "/0_template_intro/template_intro_components/template_intro.dialogue",
+		"image_path": "0_template_intro/template_intro_components/template_intro_image.png",
+	},
+	"outro":
+	{
+		"scene_path": "/4_template_outro/template_outro.tscn",
+		"components_path": "/4_template_outro/template_outro_components/",
+		"dialogue_path": "/4_template_outro/template_outro_components/template_outro.dialogue",
+	}
+}
+
+var _new_storyquest_dialog: Window
+
+
+func _enter_tree() -> void:
+	add_tool_menu_item(TOOL_MENU_LABEL, _open_new_storyquest_dialog)
+
+
+func _exit_tree() -> void:
+	remove_tool_menu_item(TOOL_MENU_LABEL)
+
+
+func _open_new_storyquest_dialog() -> void:
+	_new_storyquest_dialog = NEW_STORYQUEST_DIALOG.instantiate()
+	_new_storyquest_dialog.storyquests_path = STORYQUESTS_PATH
+	_new_storyquest_dialog.min_title_length = MIN_TITLE_LENGTH
+	_new_storyquest_dialog.create_storyquest.connect(_on_create_storyquest)
+	_new_storyquest_dialog.close_requested.connect(_close_dialog)
+	EditorInterface.popup_dialog(_new_storyquest_dialog)
+
+
+func _close_dialog() -> void:
+	_new_storyquest_dialog.queue_free()
+	_new_storyquest_dialog = null
+
+
+func _on_create_storyquest(title: String, description: String, filename: String) -> void:
+	_close_dialog()
+
+	var storyquest_path := STORYQUESTS_PATH + filename
+	assert(not FileAccess.file_exists(storyquest_path))
+
+	var error := DirAccess.make_dir_absolute(storyquest_path)
+	assert(error == OK)
+
+	# These need to be created before the following for loop, because the intro (for instance)
+	# needs to replace the outro UID in its content.
+	var storyquest_uids := {
+		"intro": ResourceUID.id_to_text(ResourceUID.create_id()),
+		"outro": ResourceUID.id_to_text(ResourceUID.create_id()),
+	}
+
+	for i in template_info:
+		var directory: String = template_info[i]["components_path"]
+		error = DirAccess.make_dir_recursive_absolute(
+			storyquest_path + directory.replacen("template_", "")
+		)
+		assert(error == OK)
+
+		var scene_path: String = template_info[i]["scene_path"].replacen("template_", "")
+		var dialogue_path: String = template_info[i]["dialogue_path"].replacen("template_", "")
+
+		var template_dialogue: DialogueResource = ResourceLoader.load(
+			TEMPLATE_PATH + template_info[i]["dialogue_path"]
+		)
```

Comment:
That's surprising because presumably DialogueManager knows how to save its resources. I think it's probably fine to do it this way for now but it'd be interesting to understand later.